File: onebot/src/main/java/cn/evole/onebot/sdk/response/group/GroupNoticeResp.java
package cn.evole.onebot.sdk.response.group;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Created on 2023/11/27.
 *
 * @author MrXiaoM
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class GroupNoticeResp {
    @SerializedName("sender_id")
    public long senderId;
    @SerializedName("publish_time")
    public long publishTime;
    @SerializedName("message")
    public Message message;

    @Data
    public static class Message {
        @SerializedName("text")
        public String text;
        @SerializedName("images")
        public List<Image> images;
    }
    @Data
    public static class Image {
        @SerializedName("id")
        public String id;
        @SerializedName("width")
        public String width;
        @SerializedName("height")
        public String height;
    }
}

File: onebot/src/main/kotlin/sdk/response/group/GroupNoticeResp.kt
package cn.evolvefield.onebot.sdk.response.group

import com.google.gson.annotations.SerializedName
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor

@Data
@AllArgsConstructor
@NoArgsConstructor
class GroupNoticeResp {
    @SerializedName("sender_id")
    var senderId = 0L

    @SerializedName("publish_time")
    var publishTime = 0L

    @SerializedName("message")
    var message: Message? = null

    @Data
    class Message {
        @SerializedName("text")
        var text: String = ""

        @SerializedName("images")
        var images: List<Image> = mutableListOf()
    }

    @Data
    class Image {
        @SerializedName("id")
        var id = ""

        @SerializedName("width")
        var width = "0"

        @SerializedName("height")
        var height = "0"
    }
}
